### DISCUSSION TITLE
Is quest "what can be left here for recycling?" working as intended?

### DISCUSSION THREAD

**Discussion by linesinalandscape** [streetcomplete/StreetComplete#6025] [Feedback]
My city is blanketed with this quest. I've found it difficult to make it work in a useful way for me, and I can see that it is causing problems for other StreetComplete mappers in the city. 

For background, municipal data about recycling containers was imported a few years ago, so recycling is generally mapped in quite a lot of detail.

Consider these two recycling containers located close to each other:
https://www.openstreetmap.org/node/9592689803 (cans, plastic_bottles, plastic_packaging - located slightly further north of the two)
https://www.openstreetmap.org/node/9592684304 (cardboard, magazines, newspaper, paper, paper_packaging)

Zoomed into the maximum in SC, I see only one green Recycling icon. If I select the quest, I see the screenshot, with a pink recycling icon immediately below the green one.
![Screenshot_20241125_093617_StreetComplete](https://github.com/user-attachments/assets/e315eca8-a1bb-4bf6-a1b0-58c35aaf4322)
 
My questions are:
- Given that both containers have recycling: tags, why is there a quest in the first place? Is it because there has not been a recent update (which would be normal, once mapped there is usually no need to do anything more unless the container moves)? If that is the reason, is it intended that SC mappers have to add multiple tags "from scratch", rather than simply confirming current data is correct?
- How would I know which of the two containers the quest relates to? I am guessing that the green and pink symbols are for the two containers, so the quest is for the one to the north, but it seems to be asking a lot of the average user to make this interpretation (they are so close that the pink icon could also be read as adding the operator name to the original quest). 

I could disable the quest, but then I wouldn't know about containers that really are missing recycling:tags. And that wouldn't solve problematic changesets by other SC mappers [(example)](https://www.openstreetmap.org/changeset/15954404) who are less aware of what is already tagged and can easily mix up adjacent containers, or not realise the containers are mapped separately or add less detailed tags than those already there.

**Comment by matkoniecz** [comment]
In general, you guessed and reasoned correctly

> Given that both containers have recycling: tags, why is there a quest in the first place? Is it because there has not been a recent update

yes

> If that is the reason, is it intended that SC mappers have to add multiple tags "from scratch", rather than simply confirming current data is correct?

yes, in the current app state

(I have a vague plan to add display of what is already mapped to make verification easier, but with how much time I have recently - it is not going any time soon)

> I am guessing that the green and pink symbols are for the two containers, so the quest is for the one to the north

yes

> but it seems to be asking a lot of the average user to make this interpretation (they are so close that the pink icon could also be read as adding the operator name to the original quest).

that is one of problems with mapping container group as separate objects, though I admit that this mapping style is valid

> I could disable the quest, but then I wouldn't know about containers that really are missing recycling:tags.

yes

you can also increase how much time SC waits before resurvey, but it will not influence other mappers

> that wouldn't solve problematic changesets by other SC mappers [(example)](https://www.openstreetmap.org/changeset/15954404) who are less aware of what is already tagged and can easily mix up adjacent containers, or not realise the containers are mapped separately or add less detailed tags than those already there.

yes, that also can happen. Usefulness of resurvey depends on how often containers move/accepted materials changed and how accurate SC mappers are. It is tricky to find correct resurvey interval and different will be optimal in a different places. So it is tricky to solve this. Some time ago I run analysis of how often situation actually changed resurvey and also had vague plan to run such analysis again, but well, see comment about lack of time. I can dig code that I used for that if anyone is interested (it was not a great code, especially obtaining data could be much smarter)

**Reply by linesinalandscape** [reply]
Thanks for the answer.

> that is one of problems with mapping container group as separate objects, though I admit that this mapping style is valid

It seems that trying to map "container groups" would just swap for a different set of problems. I can see how it would work if recycling containers are mainly in identifiable purpose-built facilities. But where containers have been randomly scattered wherever space could be found beside streets, you'd need some set of rules to define, for example, how many "groups" there are around this intersection. And other SC mappers who don't know those rules might still be confused about exactly which "group" a quest relates to. 

![image](https://github.com/user-attachments/assets/bb3c20ef-cc5a-4594-8174-c6ea73937ee8)

> Usefulness of resurvey depends on how often containers move/accepted materials changed

I guess this is partly related to the same point. If you assume that amenity=recycling main refers to "a container group", the accepted materials might change relatively often as the mix of containers changes or new ones are added. Whereas individual containers normally accept a fixed set of materials, but are more likely to move around. For these, a quest more like "does this still exist?" would probably be more useful. But I realise that OSM tagging doesn't allow single containers to be distinguished from groups. 

Overall I think that for this city at least, this quest is more likely to introduce errors than to add genuine information, In the long term the " vague plan to add display of what is already mapped" sounds like an excellent idea. But I am struggling to think of a useful suggestion for the short term.
